internal/services/network/network_manager_resource.go

```diff
@@ -175,6 +176,23 @@ func (r ManagerResource) Create() sdk.ResourceFunc {
 				return metadata.ResourceRequiresImport(r.ResourceType(), id)
 			}
 
+			if len(state.Scope[0].ManagementGroups) > 0 {
+				metadata.Logger.Info("try to register 'Microsoft.Network' provider to the Management Group scope")
+				resourceProvidersClient := metadata.Client.Resource.ResourceProvidersClient
+				for _, group := range state.Scope[0].ManagementGroups {
+					groupId, err := managementGroupParse.ManagementGroupID(group)
+					if err != nil {
+						return err
+					}
+					_, err = resourceProvidersClient.RegisterAtManagementGroupScope(ctx, "Microsoft.Network", groupId.Name)
+					if err != nil {
+						return fmt.Errorf("registering the 'Microsoft.Network' provider with Management Group %s: %+v", groupId, err)
+					}
+				}
+				// https://github.com/Azure/azure-rest-api-specs/issues/22041
+				time.Sleep(10 * time.Second)
```

Comment:
Resource Provider registration takes up to 15 minutes, and unfortunately `time.Sleep`'s aren't something we can ship/rely on - as such I think we can remove all of this RP auto-registration logic here?